include/cilantro/transform_estimation.hpp

```diff
@@ -434,4 +434,121 @@ namespace cilantro {
 
         return has_point_to_point_terms*point_to_point_correspondences.size() + has_point_to_plane_terms*point_to_plane_correspondences.size() >= Dim + 1;
     }
+
+    // Rigid, combined symmetric metric, 3D, iterative
+    // For more details about this method see
+    // Szymon Rusinkiewicz. A Symmetric Objective Function for ICP. ACM Transactions on Graphics (Proc. SIGGRAPH) 38(4), July 2019.
+    template <class TransformT, class PointCorrWeightEvaluatorT = UnityWeightEvaluator<typename TransformT::Scalar,typename TransformT::Scalar>, class PlaneCorrWeightEvaluatorT = UnityWeightEvaluator<typename TransformT::Scalar,typename TransformT::Scalar>>
+    typename std::enable_if<int(TransformT::Mode) == int(Eigen::Isometry) && TransformT::Dim == 3,bool>::type
+    estimateTransformSymmetricMetric(const ConstVectorSetMatrixMap<typename TransformT::Scalar,3> &dst_p,
+                                     const ConstVectorSetMatrixMap<typename TransformT::Scalar,3> &dst_n,
+                                     const ConstVectorSetMatrixMap<typename TransformT::Scalar,3> &src_p,
+                                     const ConstVectorSetMatrixMap<typename TransformT::Scalar,3> &src_n,
+                                     const CorrespondenceSet<typename PointCorrWeightEvaluatorT::InputScalar> &point_to_point_correspondences,
+                                     typename TransformT::Scalar point_to_point_weight,
+                                     const CorrespondenceSet<typename PlaneCorrWeightEvaluatorT::InputScalar> &point_to_plane_correspondences,
+                                     typename TransformT::Scalar point_to_plane_weight,
+                                     TransformT &tform,
+                                     size_t max_iter = 1,
+                                     typename TransformT::Scalar convergence_tol = (typename TransformT::Scalar)1e-5,
+                                     const PointCorrWeightEvaluatorT &point_corr_evaluator = PointCorrWeightEvaluatorT(),
+                                     const PlaneCorrWeightEvaluatorT &plane_corr_evaluator = PlaneCorrWeightEvaluatorT())
+    {
+        typedef typename TransformT::Scalar ScalarT;
+
+        tform.setIdentity();
+
+        const bool has_point_to_point_terms = !point_to_point_correspondences.empty() && (point_to_point_weight > (ScalarT)0.0);
+        const bool has_point_to_plane_terms = !point_to_plane_correspondences.empty() && (point_to_plane_weight > (ScalarT)0.0);
+
+        if ((!has_point_to_point_terms && !has_point_to_plane_terms) ||
+            (has_point_to_plane_terms && dst_p.cols() != dst_n.cols()))
+        {
+            return false;
+        }
+
+        const Vector<ScalarT, 3> dst_mean = dst_p.rowwise().mean();
+        const Vector<ScalarT, 3> src_mean = src_p.rowwise().mean();
+        const auto t_dst = Eigen::Translation<ScalarT, 3>(dst_mean);
+        const auto t_src = Eigen::Translation<ScalarT, 3>(-src_mean);
```

Comment:
IMHO, we should also center in the other transformation estimation methods. This improves numerical conditioning significantly. For some cases I tested, ICP diverges when the clouds are far from the origin.

Reply:
On second thought, we should center in the ICP implementations. Implementing it here will cause a significant performance hit.

Reply:
Yes, excellent point! I agree that centering at the ICP level would be a better idea performance-wise. I'll get to this (and the additional ICP method) right after I'm done moving and have my actual computer back running :D

Thanks!